recirq/quantum_chess/quantum_board.py

```diff
@@ -257,40 +278,53 @@ def sample(self, num_samples: int) -> List[int]:
             rtn = rtn + samples
         return rtn[:num_samples]
 
+    def _generate_accumulations(self, repetitions: int = 1000) -> None:
+        """ Samples the state and generates the accumulated 
+        probabilities, empty_squares, and full_squares
+        """
+        self.probabilities = [0] * 64
+        self.full_squares = (1 << 64) - 1
+        self.empty_squares = (1 << 64) - 1
+
+        samples = self.sample(repetitions)
+        for sample in samples:
+            for bit in range(64):
+                if nth_bit_of(bit, sample):
+                    self.probabilities[bit] += 1
+                    self.empty_squares = set_nth_bit(bit,self.empty_squares,0)
+                else:
+                    self.full_squares = set_nth_bit(bit,self.full_squares,0)
+
+        for bit in range(64):
+            self.probabilities[bit] = float(self.probabilities[bit]) / float(repetitions)
+        
+        self.accumulations_valid = True
```

Comment:
You need to set accumulations_valid = False in do_move.  That's why split move is failing.  It checks the probability distributions twice, and the second time it is getting the first (cached) probability distribution